docs/architecture/adr-063-privval-grpc.md

```diff
@@ -0,0 +1,75 @@
+
+The [service definition](https://grpc.io/docs/what-is-grpc/core-concepts/#service-definition) for gRPC will be defined as:
+
+```proto
+service PrivValidatorAPI {
+  rpc GetPubKey(tendermint.proto.privval.PubKeyRequest) returns (tendermint.proto.privval.PubKeyResponse);
+  rpc SignVote(tendermint.proto.privval.SignVoteRequest) returns (tendermint.proto.privval.SignedVoteResponse);
+  rpc SignProposal(tendermint.proto.privval.SignProposalRequest) returns (tendermint.proto.privval.SignedProposalResponse);
+}
+```
+
+#### Keep Alive
+
+If you have worked on the private validator system you will see that we are removing the `PingRequest` and `PingResponse` messages. These messages were used to create functionality which kept the connection alive. With gRPC there is a [keep alive feature](https://github.com/grpc/grpc/blob/master/doc/keepalive.md) that will be added along side the integration to provide the same functionality. 
+
+#### Metrics
+
+Remote signers are crucial to operating secure and consistently up Validators. In the past there were no metrics to tell the operator if something is wrong other than the node not signing. Integrating metrics into the client and provided server will be done with [prometheus](https://github.com/grpc-ecosystem/go-grpc-prometheus). This will be integrated into node's prometheus export for node operators. 
+
+#### Security
+
+[TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security) is widely adopted with the use of gRPC. There are various forms of TLS (one-way & two-way). One way is the client identifying who the server is, while two way is both parties identifying the other. For Tendermints use case having both parties identifying each other provides and extra layer of security. This requires users to generate both client and server certificates for a TLS connection. 
+
+#### Upgrade Path
+
+This is a largely breaking change for validator operators. The optimal upgrade path would be to release gRPC in a minor release, allow key management systems to migrate to the new protocol. In the next major release the current system (raw tcp/unix) is removed. This allows users to migrate to the new system and not have to coordinate upgrading the key management system alongside a network upgrade. 
+
+The upgrade of [tmkms](https://github.com/iqlusioninc/tmkms) will be coordinated with Iqlusion. They will be able to make the necessary upgrades to allow users to migrate to gRPC from the current protocol. 
+
+## Status
+
+Proposed
+
+## Consequences
+
+> This section describes the consequences, after applying the decision. All consequences should be summarized here, not just the "positive" ones.
+
+### Positive
+
+- Use an adopted standard for secure communication. (TLS)
+- Use an adopted communication protocol. (gRPC)
+- Requests are multiplexed onto the tcp connection. (http/2)
```

Comment:
Is this necessary though? aren't calls (at least signing votes and proposal) serial

Reply:
This is true. I added it because I believe some recent issues we have been observing with the privval client could have been mitigated with multiplexing. (I could be wrong)

Reply:
Okay perhaps leave it there. You might be right in regards to the `GetPubKey()` and the signing calls